src/interface/lazy.jl

```diff
@@ -7,12 +7,13 @@ const AbstractArrayOrBroadcasted = Union{AbstractArray,Broadcasted}
 mutable struct LazyTensor{T, N}
     data
     extrude::NTuple{N, Bool}
+    shape::NTuple{N, Int}
     fill_value::T
 end
-LazyTensor{T}(data, extrude::NTuple{N, Bool}, fill_value) where {T, N} = LazyTensor{T, N}(data, extrude, fill_value)
+LazyTensor{T}(data, extrude::NTuple{N, Bool}, shape::NTuple{N, Int}, fill_value) where {T, N} = LazyTensor{T, N}(data, extrude, shape, fill_value)
 
 function Base.show(io::IO, tns::LazyTensor)
-    join(io, [x ? "1" : "?" for x in tns.extrude], "×")
+    join(io, [string(x) for x in tns.shape], "×")
```

Comment:
I think you may also be able to get away with `join(io, tns.shape, "×")`, but I'm not sure, try it?